src/replay_stage.rs

```diff
@@ -180,222 +71,169 @@ impl ReplayStage {
-                                // for a signal
-                                continue;
-                            }
-                            let last_entry = blocktree
-                                .get_slot_entries(slot, meta.last_index, Some(1))
-                                .unwrap();
-                            last_entry_hash = last_entry[0].hash;
-                        }
-
-                        let old_bank = bank.clone();
-                        prev_slot = current_slot;
-                        if my_id == leader_id {
-                            // Create new bank for next slot if we are the leader for that slot
-                            bank = Self::create_and_set_working_bank(
-                                next_slot,
-                                &bank_forks,
-                                &old_bank,
-                            );
-                            current_slot = Some(next_slot);
-                            Self::reset_state(
-                                bank.ticks_per_slot(),
-                                next_slot,
-                                &mut max_tick_height_for_slot,
-                                &mut current_blob_index,
+                    // TODO: fork selection
+                    // vote on the latest one for now
+                    votable.sort();
+
+                    if let Some(latest_slot_vote) = votable.last() {
+                        let parent = bank_forks
+                            .read()
+                            .unwrap()
+                            .get(*latest_slot_vote)
+                            .unwrap()
+                            .clone();
+                        let next_slot = *latest_slot_vote + 1;
+                        let next_leader = leader_schedule_utils::slot_leader_at(next_slot, &parent);
+                        cluster_info.write().unwrap().set_leader(next_leader);
+
+                        subscriptions.notify_subscribers(&parent);
+
+                        if let Some(ref voting_keypair) = voting_keypair {
+                            let keypair = voting_keypair.as_ref();
+                            let vote = VoteTransaction::new_vote(
+                                keypair,
+                                *latest_slot_vote,
+                                parent.last_id(),
+                                0,
                             );
-                        } else {
-                            current_slot = None;
+                            cluster_info.write().unwrap().push_vote(vote);
                         }
-
-                        if leader_id != current_leader_id {
-                            // TODO: Remove this soon once we boot the leader from ClusterInfo
-                            cluster_info.write().unwrap().set_leader(leader_id);
+                        if next_leader == my_id {
+                            let mut wforks = bank_forks.write().unwrap();
```

Comment:
can you make this a one-liner?  these wXXX or rXXX bug my eyes...

`bank_forks.write().unwrap().insert(tpu_bank);`